platform/mds/mds/src/main/java/org/motechproject/mds/service/DefaultMotechDataService.java

```diff
@@ -475,8 +475,13 @@ protected InstanceSecurityRestriction validateCredentials(T instance) {
         return restriction;
     }
 
-    protected void verifySchemaVersion(Object record, Long recordId) {
-        String schemaField = HistoryTrashClassHelper.schemaVersion(record.getClass());
+    protected void verifySchemaVersion(Object record, Long recordId, boolean forTrash) {
+        String schemaField;
+        if (forTrash) {
+            schemaField = "schemaVersion";
```

Comment:
why is this different for trash and history? "schemaVersion" is also a consatant somewhere by the way